### ISSUE TITLE
findOneAndUpdate returning doc BEFORE update with 4.0.x

### ISSUE THREAD

**Issue by christinedraper** [Automattic/mongoose#2994]
findOneAndUpdate(query, update, callback) with 3.8.21 returns the document AFTER update in the callback. With 4.0.x it returns the document BEFORE update.

In the example below, this is the result when running the test three times starting with an empty SystemCounters collection and using 3.8.21:

{ _id: 'Test', sequence: 1}
{ _id: 'Test', sequence: 2}
{ _id: 'Test', sequence: 3}

This is what happens with 4.0.3:
null
{ _id: 'Test', sequence: 1}
{ _id: 'Test', sequence: 2}

```
var mongoose = require('mongoose');

var SystemCountersSchema = new mongoose.Schema({
	_id: { type: String, required: true },
	sequence: { type: Number, required: true },
},{
	versionKey: false,
	collection : 'SystemCounters'
}
);

var conn = mongoose.createConnection();
conn.open("mongodb://localhost/MyDB", {}, function(){
	var systemCounters = conn.model('SystemCounters', SystemCountersSchema);
	systemCounters.findOneAndUpdate({_id: "Test"}, {$inc: {sequence: 1}}, {upsert: true}, function(err, counter)
		{
			if (err) {
				console.log("Error incrementing system counter for ", name);
			} else{
				console.log("Counter is ", counter);
			}
			conn.close();
		});
	
});
```

**Comment by christinedraper** [comment]
Closed as duplicate of #2994 and working as designed

**Comment by vkarpov15** [comment]
See [4.0 release notes](https://github.com/Automattic/mongoose/wiki/4.0-Release-Notes) for rationale behind this change.
